Camera2SlowMotion/app/src/main/java/com/example/android/camera2/slowmo/MainActivity.java

```diff
@@ -205,60 +211,45 @@ public void onFailure(@NonNull Exception e) {
 
   private void parseOCR() {
     try {
-      String last = resultsOCR.get(0);
-      Integer recordStartTime =
-          convertStringTimeToInt(CameraFragment.Companion.getRecordingStartTime());
-      tmpOCR.add(0);
-      deltaOCR.add(recordStartTime);
-      Integer offset = 1000 / fps;
-      for (int i = 1; i < resultsOCR.size(); i++) {
-        if (!last.equals(resultsOCR.get(i))) {
-          last = resultsOCR.get(i);
-          tmpOCR.add(i);
-          deltaOCR.add(recordStartTime + (i * offset));
-          Log.d(tag, "In parseOCR:" + i + "  " + (recordStartTime + (i * offset)));
-        }
+      Long recordStartTime = CameraFragment.Companion.getRecordingStartMillis();
+      Long offset = 1000L / fps;
+      for (int i = 0; i < resultsOCR.size(); i++) {
+        deltaOCR.add(recordStartTime + (i * offset));
       }
     } catch (Exception e) {
       Log.d("Rokus Logs:", "Parsing OCR failed with: " + e.getMessage());
     }
   }
 
-  // Parse timestamps received from server. Last 12 characters of time is important here.
+  // Parse timestamps received from server.
   private void parseServer() {
-    Integer length = serverCache.get(0).length() - 12;
     for (int i = 0; i < serverCache.size(); i++) {
```

Comment:
optional: you could use a map here instead.  Both ways are fine, but if you don't know about map in Java you should learn.